sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementSharesExtensions.cs

```diff
@@ -110,25 +114,46 @@ public static FileSmbProperties GetFileSmbProperties(
                     : properties?.RawProperties?.TryGetValue(DataMovementConstants.ResourceProperties.LastWrittenOn, out object fileLastWrittenOn) == true
                         ? (DateTimeOffset?)fileLastWrittenOn
                         : default,
-                FileChangedOn = (options?._isFileChangedOnSet ?? false)
-                    ? options?.FileChangedOn
-                    : properties?.RawProperties?.TryGetValue(DataMovementConstants.ResourceProperties.ChangedOnTime, out object fileChangedOn) == true
-                        ? (DateTimeOffset?)fileChangedOn
-                        : default,
+                FileChangedOn = options?.Nfs == true
+                    ? default
+                    : (options?._isFileChangedOnSet ?? false)
+                        ? options?.FileChangedOn
+                        : properties?.RawProperties?.TryGetValue(DataMovementConstants.ResourceProperties.ChangedOnTime, out object fileChangedOn) == true
+                            ? (DateTimeOffset?)fileChangedOn
+                            : default,
             };
         }
 
+        public static FilePosixProperties GetFilePosixProperties(
+            this ShareFileStorageResourceOptions options,
+            NfsFileMode destinationFileMode,
+            string destinationOwner,
+            string destinationGroup,
+            NfsFileType? destinationFileType,
+            long? destinationLinkCount)
+        {
+            if (options?.Nfs != true)
```

Comment:
Testing if an optional bool is `!= true` is confusing to me. I had to actually write out the below table to reason over what was happening.

| options?.Nfs | options?.Nfs != true |
| ------ | ------ |
| true | false |
| false | true |
| default | true |

I feel like the following is much more readable:

```c#
if (options?.Nfs)
{
    return FilesModelFactory.FilePosixProperties(...)
}
return new();
```